homotopy-common/src/tree.rs

```diff
@@ -120,19 +119,27 @@ impl<T> Tree<T> {
     }
 
     #[inline]
-    pub fn with<F, U>(&self, node: Node, f: F) -> U
+    pub fn with<F, U>(&self, node: Node, f: F) -> Option<U>
     where
         F: FnOnce(&NodeData<T>) -> U,
     {
-        f(&self.nodes[node])
+        if node.0 < self.nodes.len() {
```

Comment:
Guarding against index out of bounds errors feel a bit unnecessary to me.

Reply:
Yeah those maybe were a bit excessive, but plenty of other places were calling into it without doing any bounds checking.